onnxruntime/test/framework/bfc_arena_test.cc

```diff
@@ -374,5 +390,20 @@ TEST(StreamAwareArenaTest, TwoStreamAllocation) {
   a.Free(stream2_chunk_f);
 }
 
+TEST(StreamAwareArenaTest, TestSecureTheChunk) {
+  StreamAwareArena a(std::unique_ptr<IAllocator>(new CPUAllocator()), 1 << 30, true);
+  OrtDevice tmp;
+  StreamMock stream1(tmp), stream2(tmp);
+
+  void* p1 = a.AllocOnStream(BFCArena::DEFAULT_INITIAL_CHUNK_SIZE_BYTES, &stream1, nullptr);
```

Comment:
could we provide a mock for the wait function, to test the wait function is invoked as expected?